### ISSUE TITLE
Attempting to call or adjust the <player.adventurer.save> attribute returns a number of errors

### ISSUE THREAD

**Issue by Rickyling97** [navett52/Argonauts#5]
In this case, I attemped two methods of calling the attribute.
The first was by calling it as a variable.
`
14:47:07] [Server thread/INFO]: Rickyling issued server command: /ex announce <player.adventurer.save>
[14:47:07] [pool-23-thread-1/ERROR]: [Argonauts] Unable to retreive connection
java.sql.SQLException: database connection closed
	at org.sqlite.core.CoreConnection.checkOpen(CoreConnection.java:336) ~[spigot.jar:git-Spigot-99c2a37-d35483b]
	at org.sqlite.jdbc4.JDBC4Connection.prepareStatement(JDBC4Connection.java:45) ~[spigot.jar:git-Spigot-99c2a37-d35483b]
	at org.sqlite.jdbc3.JDBC3Connection.prepareStatement(JDBC3Connection.java:263) ~[spigot.jar:git-Spigot-99c2a37-d35483b]
	at org.sqlite.jdbc3.JDBC3Connection.prepareStatement(JDBC3Connection.java:235) ~[spigot.jar:git-Spigot-99c2a37-d35483b]
	at com.phenom.argonauts.database.ArgonautsDatabase.saveAdventurer(ArgonautsDatabase.java:135) [Argonauts.jar:?]
	at com.phenom.argonauts.denizens.dAdventurer.getAttribute(dAdventurer.java:111) [Argonauts.jar:?]
	at com.phenom.argonauts.denizens.dPlayerExt.getAttribute(dPlayerExt.java:36) [Argonauts.jar:?]
	at net.aufdemrand.denizen.objects.dPlayer.getAttribute(dPlayer.java:2160) [denizen-1.0.2-SNAPSHOT.jar:?]
	at net.aufdemrand.denizen.tags.core.PlayerTags.playerTags(PlayerTags.java:90) [denizen-1.0.2-SNAPSHOT.jar:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_91]
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_91]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_91]
	at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_91]
	at net.aufdemrand.denizencore.tags.TagManager.fireEvent(TagManager.java:83) [denizen-1.0.2-SNAPSHOT.jar:?]
	at net.aufdemrand.denizencore.tags.TagManager$1.run(TagManager.java:258) [denizen-1.0.2-SNAPSHOT.jar:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source) [?:1.8.0_91]
	at java.util.concurrent.FutureTask.run(Unknown Source) [?:1.8.0_91]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source) [?:1.8.0_91]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source) [?:1.8.0_91]
	at java.lang.Thread.run(Unknown Source) [?:1.8.0_91]
[14:47:07] [pool-23-thread-1/ERROR]: [Argonauts] Failed to close MySQL connection: 
org.sqlite.SQLiteException: [SQLITE_ERROR] SQL error or missing database (Connection is closed)
	at org.sqlite.core.DB.newSQLException(DB.java:909) ~[spigot.jar:git-Spigot-99c2a37-d35483b]
	at org.sqlite.core.CoreStatement.internalClose(CoreStatement.java:115) ~[spigot.jar:git-Spigot-99c2a37-d35483b]
	at org.sqlite.jdbc3.JDBC3Statement.close(JDBC3Statement.java:35) ~[spigot.jar:git-Spigot-99c2a37-d35483b]
	at org.sqlite.jdbc4.JDBC4Statement.close(JDBC4Statement.java:27) ~[spigot.jar:git-Spigot-99c2a37-d35483b]
	at com.phenom.argonauts.database.ArgonautsDatabase.close(ArgonautsDatabase.java:384) [Argonauts.jar:?]
	at com.phenom.argonauts.database.ArgonautsDatabase.saveAdventurer(ArgonautsDatabase.java:151) [Argonauts.jar:?]
	at com.phenom.argonauts.denizens.dAdventurer.getAttribute(dAdventurer.java:111) [Argonauts.jar:?]
	at com.phenom.argonauts.denizens.dPlayerExt.getAttribute(dPlayerExt.java:36) [Argonauts.jar:?]
	at net.aufdemrand.denizen.objects.dPlayer.getAttribute(dPlayer.java:2160) [denizen-1.0.2-SNAPSHOT.jar:?]
	at net.aufdemrand.denizen.tags.core.PlayerTags.playerTags(PlayerTags.java:90) [denizen-1.0.2-SNAPSHOT.jar:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_91]
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_91]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_91]
	at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_91]
	at net.aufdemrand.denizencore.tags.TagManager.fireEvent(TagManager.java:83) [denizen-1.0.2-SNAPSHOT.jar:?]
	at net.aufdemrand.denizencore.tags.TagManager$1.run(TagManager.java:258) [denizen-1.0.2-SNAPSHOT.jar:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source) [?:1.8.0_91]
	at java.util.concurrent.FutureTask.run(Unknown Source) [?:1.8.0_91]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source) [?:1.8.0_91]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source) [?:1.8.0_91]
	at java.lang.Thread.run(Unknown Source) [?:1.8.0_91]
[14:47:07] [pool-23-thread-1/ERROR]: [Argonauts] Unable to retreive connection
java.sql.SQLException: database connection closed
	at org.sqlite.core.CoreConnection.checkOpen(CoreConnection.java:336) ~[spigot.jar:git-Spigot-99c2a37-d35483b]
	at org.sqlite.jdbc4.JDBC4Connection.prepareStatement(JDBC4Connection.java:45) ~[spigot.jar:git-Spigot-99c2a37-d35483b]
	at org.sqlite.jdbc3.JDBC3Connection.prepareStatement(JDBC3Connection.java:263) ~[spigot.jar:git-Spigot-99c2a37-d35483b]
	at org.sqlite.jdbc3.JDBC3Connection.prepareStatement(JDBC3Connection.java:235) ~[spigot.jar:git-Spigot-99c2a37-d35483b]
	at com.phenom.argonauts.database.ArgonautsDatabase.saveAdventurer(ArgonautsDatabase.java:156) [Argonauts.jar:?]
	at com.phenom.argonauts.denizens.dAdventurer.getAttribute(dAdventurer.java:111) [Argonauts.jar:?]
	at com.phenom.argonauts.denizens.dPlayerExt.getAttribute(dPlayerExt.java:36) [Argonauts.jar:?]
	at net.aufdemrand.denizen.objects.dPlayer.getAttribute(dPlayer.java:2160) [denizen-1.0.2-SNAPSHOT.jar:?]
	at net.aufdemrand.denizen.tags.core.PlayerTags.playerTags(PlayerTags.java:90) [denizen-1.0.2-SNAPSHOT.jar:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_91]
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_91]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_91]
	at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_91]
	at net.aufdemrand.denizencore.tags.TagManager.fireEvent(TagManager.java:83) [denizen-1.0.2-SNAPSHOT.jar:?]
	at net.aufdemrand.denizencore.tags.TagManager$1.run(TagManager.java:258) [denizen-1.0.2-SNAPSHOT.jar:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source) [?:1.8.0_91]
	at java.util.concurrent.FutureTask.run(Unknown Source) [?:1.8.0_91]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source) [?:1.8.0_91]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source) [?:1.8.0_91]
	at java.lang.Thread.run(Unknown Source) [?:1.8.0_91]
[14:47:07] [pool-23-thread-1/ERROR]: [Argonauts] Failed to close MySQL connection: 
org.sqlite.SQLiteException: [SQLITE_ERROR] SQL error or missing database (Connection is closed)
	at org.sqlite.core.DB.newSQLException(DB.java:909) ~[spigot.jar:git-Spigot-99c2a37-d35483b]
	at org.sqlite.core.CoreStatement.internalClose(CoreStatement.java:115) ~[spigot.jar:git-Spigot-99c2a37-d35483b]
	at org.sqlite.jdbc3.JDBC3Statement.close(JDBC3Statement.java:35) ~[spigot.jar:git-Spigot-99c2a37-d35483b]
	at org.sqlite.jdbc4.JDBC4Statement.close(JDBC4Statement.java:27) ~[spigot.jar:git-Spigot-99c2a37-d35483b]
	at com.phenom.argonauts.database.ArgonautsDatabase.close(ArgonautsDatabase.java:384) [Argonauts.jar:?]
	at com.phenom.argonauts.database.ArgonautsDatabase.saveAdventurer(ArgonautsDatabase.java:169) [Argonauts.jar:?]
	at com.phenom.argonauts.denizens.dAdventurer.getAttribute(dAdventurer.java:111) [Argonauts.jar:?]
	at com.phenom.argonauts.denizens.dPlayerExt.getAttribute(dPlayerExt.java:36) [Argonauts.jar:?]
	at net.aufdemrand.denizen.objects.dPlayer.getAttribute(dPlayer.java:2160) [denizen-1.0.2-SNAPSHOT.jar:?]
	at net.aufdemrand.denizen.tags.core.PlayerTags.playerTags(PlayerTags.java:90) [denizen-1.0.2-SNAPSHOT.jar:?]
	at sun.reflect.NativeMethodAccessorImpl.invoke0(Native Method) ~[?:1.8.0_91]
	at sun.reflect.NativeMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_91]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_91]
	at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_91]
	at net.aufdemrand.denizencore.tags.TagManager.fireEvent(TagManager.java:83) [denizen-1.0.2-SNAPSHOT.jar:?]
	at net.aufdemrand.denizencore.tags.TagManager$1.run(TagManager.java:258) [denizen-1.0.2-SNAPSHOT.jar:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source) [?:1.8.0_91]
	at java.util.concurrent.FutureTask.run(Unknown Source) [?:1.8.0_91]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source) [?:1.8.0_91]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source) [?:1.8.0_91]
	at java.lang.Thread.run(Unknown Source) [?:1.8.0_91]
[14:47:07] [Server thread/INFO]: null[m
`

The second method I attempted to directly adjust it.
`
[14:47:40] [Server thread/INFO]: Rickyling issued server command: /ex adjust <player.adventurer.save> 0
[14:47:40] [pool-24-thread-1/ERROR]: [Argonauts] Unable to retreive connection
java.sql.SQLException: database connection closed
	at org.sqlite.core.CoreConnection.checkOpen(CoreConnection.java:336) ~[spigot.jar:git-Spigot-99c2a37-d35483b]
	at org.sqlite.jdbc4.JDBC4Connection.prepareStatement(JDBC4Connection.java:45) ~[spigot.jar:git-Spigot-99c2a37-d35483b]
	at org.sqlite.jdbc3.JDBC3Connection.prepareStatement(JDBC3Connection.java:263) ~[spigot.jar:git-Spigot-99c2a37-d35483b]
	at org.sqlite.jdbc3.JDBC3Connection.prepareStatement(JDBC3Connection.java:235) ~[spigot.jar:git-Spigot-99c2a37-d35483b]
	at com.phenom.argonauts.database.ArgonautsDatabase.saveAdventurer(ArgonautsDatabase.java:135) [Argonauts.jar:?]
	at com.phenom.argonauts.denizens.dAdventurer.getAttribute(dAdventurer.java:111) [Argonauts.jar:?]
	at com.phenom.argonauts.denizens.dPlayerExt.getAttribute(dPlayerExt.java:36) [Argonauts.jar:?]
	at net.aufdemrand.denizen.objects.dPlayer.getAttribute(dPlayer.java:2160) [denizen-1.0.2-SNAPSHOT.jar:?]
	at net.aufdemrand.denizen.tags.core.PlayerTags.playerTags(PlayerTags.java:90) [denizen-1.0.2-SNAPSHOT.jar:?]
	at sun.reflect.GeneratedMethodAccessor34.invoke(Unknown Source) ~[?:?]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_91]
	at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_91]
	at net.aufdemrand.denizencore.tags.TagManager.fireEvent(TagManager.java:83) [denizen-1.0.2-SNAPSHOT.jar:?]
	at net.aufdemrand.denizencore.tags.TagManager$1.run(TagManager.java:258) [denizen-1.0.2-SNAPSHOT.jar:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source) [?:1.8.0_91]
	at java.util.concurrent.FutureTask.run(Unknown Source) [?:1.8.0_91]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source) [?:1.8.0_91]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source) [?:1.8.0_91]
	at java.lang.Thread.run(Unknown Source) [?:1.8.0_91]
[14:47:40] [pool-24-thread-1/ERROR]: [Argonauts] Failed to close MySQL connection: 
org.sqlite.SQLiteException: [SQLITE_ERROR] SQL error or missing database (Connection is closed)
	at org.sqlite.core.DB.newSQLException(DB.java:909) ~[spigot.jar:git-Spigot-99c2a37-d35483b]
	at org.sqlite.core.CoreStatement.internalClose(CoreStatement.java:115) ~[spigot.jar:git-Spigot-99c2a37-d35483b]
	at org.sqlite.jdbc3.JDBC3Statement.close(JDBC3Statement.java:35) ~[spigot.jar:git-Spigot-99c2a37-d35483b]
	at org.sqlite.jdbc4.JDBC4Statement.close(JDBC4Statement.java:27) ~[spigot.jar:git-Spigot-99c2a37-d35483b]
	at com.phenom.argonauts.database.ArgonautsDatabase.close(ArgonautsDatabase.java:384) [Argonauts.jar:?]
	at com.phenom.argonauts.database.ArgonautsDatabase.saveAdventurer(ArgonautsDatabase.java:151) [Argonauts.jar:?]
	at com.phenom.argonauts.denizens.dAdventurer.getAttribute(dAdventurer.java:111) [Argonauts.jar:?]
	at com.phenom.argonauts.denizens.dPlayerExt.getAttribute(dPlayerExt.java:36) [Argonauts.jar:?]
	at net.aufdemrand.denizen.objects.dPlayer.getAttribute(dPlayer.java:2160) [denizen-1.0.2-SNAPSHOT.jar:?]
	at net.aufdemrand.denizen.tags.core.PlayerTags.playerTags(PlayerTags.java:90) [denizen-1.0.2-SNAPSHOT.jar:?]
	at sun.reflect.GeneratedMethodAccessor34.invoke(Unknown Source) ~[?:?]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_91]
	at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_91]
	at net.aufdemrand.denizencore.tags.TagManager.fireEvent(TagManager.java:83) [denizen-1.0.2-SNAPSHOT.jar:?]
	at net.aufdemrand.denizencore.tags.TagManager$1.run(TagManager.java:258) [denizen-1.0.2-SNAPSHOT.jar:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source) [?:1.8.0_91]
	at java.util.concurrent.FutureTask.run(Unknown Source) [?:1.8.0_91]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source) [?:1.8.0_91]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source) [?:1.8.0_91]
	at java.lang.Thread.run(Unknown Source) [?:1.8.0_91]
[14:47:40] [pool-24-thread-1/ERROR]: [Argonauts] Unable to retreive connection
java.sql.SQLException: database connection closed
	at org.sqlite.core.CoreConnection.checkOpen(CoreConnection.java:336) ~[spigot.jar:git-Spigot-99c2a37-d35483b]
	at org.sqlite.jdbc4.JDBC4Connection.prepareStatement(JDBC4Connection.java:45) ~[spigot.jar:git-Spigot-99c2a37-d35483b]
	at org.sqlite.jdbc3.JDBC3Connection.prepareStatement(JDBC3Connection.java:263) ~[spigot.jar:git-Spigot-99c2a37-d35483b]
	at org.sqlite.jdbc3.JDBC3Connection.prepareStatement(JDBC3Connection.java:235) ~[spigot.jar:git-Spigot-99c2a37-d35483b]
	at com.phenom.argonauts.database.ArgonautsDatabase.saveAdventurer(ArgonautsDatabase.java:156) [Argonauts.jar:?]
	at com.phenom.argonauts.denizens.dAdventurer.getAttribute(dAdventurer.java:111) [Argonauts.jar:?]
	at com.phenom.argonauts.denizens.dPlayerExt.getAttribute(dPlayerExt.java:36) [Argonauts.jar:?]
	at net.aufdemrand.denizen.objects.dPlayer.getAttribute(dPlayer.java:2160) [denizen-1.0.2-SNAPSHOT.jar:?]
	at net.aufdemrand.denizen.tags.core.PlayerTags.playerTags(PlayerTags.java:90) [denizen-1.0.2-SNAPSHOT.jar:?]
	at sun.reflect.GeneratedMethodAccessor34.invoke(Unknown Source) ~[?:?]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_91]
	at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_91]
	at net.aufdemrand.denizencore.tags.TagManager.fireEvent(TagManager.java:83) [denizen-1.0.2-SNAPSHOT.jar:?]
	at net.aufdemrand.denizencore.tags.TagManager$1.run(TagManager.java:258) [denizen-1.0.2-SNAPSHOT.jar:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source) [?:1.8.0_91]
	at java.util.concurrent.FutureTask.run(Unknown Source) [?:1.8.0_91]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source) [?:1.8.0_91]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source) [?:1.8.0_91]
	at java.lang.Thread.run(Unknown Source) [?:1.8.0_91]
[14:47:40] [pool-24-thread-1/ERROR]: [Argonauts] Failed to close MySQL connection: 
org.sqlite.SQLiteException: [SQLITE_ERROR] SQL error or missing database (Connection is closed)
	at org.sqlite.core.DB.newSQLException(DB.java:909) ~[spigot.jar:git-Spigot-99c2a37-d35483b]
	at org.sqlite.core.CoreStatement.internalClose(CoreStatement.java:115) ~[spigot.jar:git-Spigot-99c2a37-d35483b]
	at org.sqlite.jdbc3.JDBC3Statement.close(JDBC3Statement.java:35) ~[spigot.jar:git-Spigot-99c2a37-d35483b]
	at org.sqlite.jdbc4.JDBC4Statement.close(JDBC4Statement.java:27) ~[spigot.jar:git-Spigot-99c2a37-d35483b]
	at com.phenom.argonauts.database.ArgonautsDatabase.close(ArgonautsDatabase.java:384) [Argonauts.jar:?]
	at com.phenom.argonauts.database.ArgonautsDatabase.saveAdventurer(ArgonautsDatabase.java:169) [Argonauts.jar:?]
	at com.phenom.argonauts.denizens.dAdventurer.getAttribute(dAdventurer.java:111) [Argonauts.jar:?]
	at com.phenom.argonauts.denizens.dPlayerExt.getAttribute(dPlayerExt.java:36) [Argonauts.jar:?]
	at net.aufdemrand.denizen.objects.dPlayer.getAttribute(dPlayer.java:2160) [denizen-1.0.2-SNAPSHOT.jar:?]
	at net.aufdemrand.denizen.tags.core.PlayerTags.playerTags(PlayerTags.java:90) [denizen-1.0.2-SNAPSHOT.jar:?]
	at sun.reflect.GeneratedMethodAccessor34.invoke(Unknown Source) ~[?:?]
	at sun.reflect.DelegatingMethodAccessorImpl.invoke(Unknown Source) ~[?:1.8.0_91]
	at java.lang.reflect.Method.invoke(Unknown Source) ~[?:1.8.0_91]
	at net.aufdemrand.denizencore.tags.TagManager.fireEvent(TagManager.java:83) [denizen-1.0.2-SNAPSHOT.jar:?]
	at net.aufdemrand.denizencore.tags.TagManager$1.run(TagManager.java:258) [denizen-1.0.2-SNAPSHOT.jar:?]
	at java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source) [?:1.8.0_91]
	at java.util.concurrent.FutureTask.run(Unknown Source) [?:1.8.0_91]
	at java.util.concurrent.ThreadPoolExecutor.runWorker(Unknown Source) [?:1.8.0_91]
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(Unknown Source) [?:1.8.0_91]
	at java.lang.Thread.run(Unknown Source) [?:1.8.0_91]
`

**Comment by navett52** [comment]
I think I figured out this issue already. I made a convenience method to close connections and query statements after they're used. With updates I have to update each table separately so I can't do a mega query with tons of joins, so basically I had multiple query statements within a method. After the first one, the connection gets closed and I never re-open it. I think that's the problem so I can probably fix that tomorrow.
